src/ResourceManagement/Sql/SqlDatabaseUsageMetricImpl.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+namespace Microsoft.Azure.Management.Sql.Fluent
+{
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using Microsoft.Azure.Management.Sql.Fluent.Models;
+    using System;
+
+    /// <summary>
+    /// Implementation for Azure SQL Database usage.
+    /// </summary>
+///GENTHASH:Y29tLm1pY3Jvc29mdC5henVyZS5tYW5hZ2VtZW50LnNxbC5pbXBsZW1lbnRhdGlvbi5TcWxEYXRhYmFzZVVzYWdlTWV0cmljSW1wbA==
+    internal partial class SqlDatabaseUsageMetricImpl  :
+        Wrapper<Models.DatabaseUsage>,
+        ISqlDatabaseUsageMetric
+    {
+        ///GENMHASH:A0159F79F51C286DD71841E95366B32C:C0C35E00AF4E17F141675A2C05C7067B
+        internal SqlDatabaseUsageMetricImpl(Models.DatabaseUsage innerObject)
+            : base(innerObject)
+        {
+        }
+
+        ///GENMHASH:98D67B93923AC46ECFE338C62748BCCB:AE6A688E48A33F4836A5CFB695421894
+        public string Unit()
+        {
+            return this.Inner.Unit;
+        }
+
+        ///GENMHASH:19FB5490B29F08AC39628CD5F893E975:D646459DD47DA53CB973DA0F86C056D7
+        public string DisplayName()
+        {
+            return this.Inner.DisplayName;
+        }
+
+        ///GENMHASH:3E38805ED0E7BA3CAEE31311D032A21C:61C1065B307679F3800C701AE0D87070
+        public string Name()
+        {
+            return this.Inner.Name;
+        }
+
+        ///GENMHASH:9D196E486CC1E35756FD0BEDAB3F3BE4:1A97C3177E4C3DBBD400096F05CC2999
+        public double Limit()
+        {
+            return this.Inner.Limit.GetValueOrDefault(0.0d);
```

Comment:
>            return this.Inner.Limit.GetValueOrDefault(0.0d) [](start = 0, length = 59)

my comment about having nullible in the wrapper is better shown here - so if I'd not set value for Limit during create you will return me 0.0 after create?

Reply:
You cannot set a "limit" here; it's a read only property which should be consistent with what we return in Java.